### PR TITLE
Fix response time display error when not 2xx

### PR DESCRIPTION
**Pull request by sapiderman**
# Monika Pull Request 

## What feature/issue does this PR fix
1. Closing #313 

## How did you implement / how did you fix it  
1.  The response time data lives in the config.extraData rather than than in error.response.
2. So returning config.extraData instead when not 2xx
3. Add some documentation and update the sanitize event definition

## How to test  
1. see Issue #313 for how to reproduce

## Screenshot before fix vs after fix
image

### CODE REVIEW

File: src/components/probe/probing.ts [hunk 1]
```diff
@@ -66,31 +66,28 @@ export async function probing(
...
 
     if (error.response) {
       // Axios doesn't always return error response
-      errStatus = error.response.status
+      errResponseCode = error.response.status
       errData = error.response.data
       errHdr = error.response.headers
     } else {
-      errStatus = 500 // TODO: how to detect timeouts?
+      errResponseCode = 500 // TODO: how to detect timeouts?
```
Comment (gsukanto) [inline src/components/probe/probing.ts:79]: For detecting timeouts error, we can use:
```
if (error.code === 'ECONNABORTED')
```

**Review by gsukanto — APPROVED** [review]
lgtm

File: src/components/probe/probing.ts (hunk 1 above)
Comment (gsukanto) [reply src/components/probe/probing.ts:79]: reff: https://github.com/axios/axios/blob/v0.21.1/lib/adapters/http.js#L280

**Review by armedi — APPROVED** [review]

File: src/components/probe/probing.ts (hunk 1 above)
Comment (sapiderman) [reply src/components/probe/probing.ts:79]: thanks greg.. yup found this

**Review by nicnocquee — APPROVED** [review]

**Pull request merged**